host-controller/src/main/java/org/jboss/as/host/controller/operations/StartServersHandler.java

```diff
@@ -112,6 +118,11 @@ private void cleanStartServers(final ModelNode servers, final ModelNode domainMo
         Map<String, ProcessInfo> processInfos = serverInventory.determineRunningProcesses();
         for(final Property serverProp : servers.asPropertyList()) {
             String serverName = serverProp.getName();
+
+            // WFCORE-3897 resolve required server group early no matter if server is auto-started
+            String serverGroupName = serverProp.getValue().get(GROUP).asString();
+            resolveExpressions(domainModel.require(SERVER_GROUP).require(serverGroupName), expressionResolver, true);
```

Comment:
@soul2zimate I don't understand your patch.

Do you need to resolve the expressions of the `server-groups`?
It seems to me, you only need to check for the presence of a `server-group` resource named `serverGroupName`.

Why are you also resolving its expressions?

Reply:
You're right, it does not need to be expression resolved to check its value. Please hold this, there is a comment in WFCORE-3897 about detecting the missing server-group by capability validation once domain model is available(such validation is currently ignored as explanation at https://github.com/wildfly/wildfly-core/blob/5.0.0.Final/controller/src/main/java/org/jboss/as/controller/CapabilityRegistry.java#L860), I am investigating on it.